File: oc/UITableViewDemo/UITableViewDemo/Car.m
//
//  Car.m
//  UITableViewDemo
//
//  Created by dfang on 2017-11-22.
//  Copyright © 2017年 com.east. All rights reserved.
//

#import "Car.h"

@implementation Car

- (instancetype)initWithDic:(NSDictionary *)dic {
    if (self = [super init]) {
        [self setValuesForKeysWithDictionary:dic];
    }
    return self;
}

+ (instancetype)carWithDic:(NSDictionary *)dic {
    return [[Car alloc] initWithDic:dic];
}

@end

File: swift/UITableViewDemo/UITableViewDemo/Car.swift
//
//  Car.swift
//  UITableViewDemo
//
//  Created by dfang on 2017-11-21.
//  Copyright © 2017年 com.east. All rights reserved.
//

import Foundation

class Car: NSObject {
    var name: String?
    var icon: String?
    
    init(dict: NSDictionary) {
        name = dict.value(forKey: "name") as? String ?? ""
        icon = dict.value(forKey: "icon") as? String ?? ""
    }
}
